nb2workflow/deploy.py

```diff
@@ -230,40 +224,51 @@ def _build_with_kaniko(git_origin,
                       restartPolicy: Never
                 """))
         
-        sp.check_call([
-            "kubectl",
-            "create",
-            "-f",
-            "buildjob.yaml"
-        ], cwd=tmpdir)
-        
-        sp.check_call([
-            "kubectl",
-            "-n",
-            f"{namespace}",
-            "wait",
-            "--for=condition=complete",
-            "--timeout=30m",
-            f"job/kaniko-build-{suffix}"
-        ])
-        
-        if cleanup:
+        try:
             sp.check_call([
                 "kubectl",
-                "-n",
-                f"{namespace}",
-                "delete",
-                f"job/kaniko-build-{suffix}"
-            ])
+                "create",
+                "configmap",
+                "-n", namespace,
+                f"nb2w-dockerfile-{suffix}",
+                "--from-file=Dockerfile=Dockerfile"
+            ], cwd=tmpdir)
+                    
+            sp.check_call([
+                "kubectl",
+                "create",
+                "-f",
+                "buildjob.yaml"
+            ], cwd=tmpdir)
             
             sp.check_call([
                 "kubectl",
                 "-n",
                 f"{namespace}",
-                "delete",
-                "configmap",
-                f"nb2w-dockerfile-{suffix}"
+                "wait",
+                "--for=condition=complete",
+                "--timeout=120m",
+                f"job/kaniko-build-{suffix}"
             ])
+            
```

Comment:
maybe we want some treatment if there is an issue, a message, an email, gitlab job update?

Reply:
This is done on the bot side

Reply:
but here it will exit without error?..

Reply:
The idea was that `kubectl wait` will return nonzero exit status if the job fails. 
 I used to believe that it is returned as soon as backoffLimit is exceeded in job. But actually, it only terminates on wait timeout, which will work, but is definitely not what I wanted.

 I explored the possibilities, it seems that [this advice](https://saturncloud.io/blog/is-there-a-better-way-to-wait-for-kubernetes-job-completion-a-deep-dive-into-kubernetes-job-monitoring/) is a way to go. Will update